Gems/Atom/Feature/Common/Assets/Shaders/PostProcessing/EditorModeMask.azsl

```diff
@@ -35,45 +56,60 @@ struct VSOutput
     // Base fields (required by the template azsli file)...
     // "centroid" is needed for SV_Depth to compile
     precise linear centroid float4 m_position : SV_Position;
+
+    // Fragment depth
+    float m_depth : UV0;
 };
 
 VSOutput MainVS(VSInput IN)
 {
-    VSOutput OUT;    
-    float3 worldPosition = mul(ObjectSrg::GetWorldMatrix(), float4(IN.m_position, 1.0)).xyz;
-    OUT.m_position = mul(ViewSrg::m_viewProjectionMatrix, float4(worldPosition, 1.0));
-
-    // Offset the depth of redrawn entities to avoid z fighting with the underlying entity rendered data
-    // A proper programatic solution will be investigated in LYN-10304
-    OUT.m_position.z += 0.001;
-
-/*
-    For use in LYN-9929
+    VSOutput OUT;
 
-    // NDCs
-    float3 ndc = float3(OUT.m_position.x / OUT.m_position.w, OUT.m_position. y/ OUT.m_position.w, OUT.m_position.z / OUT.m_position.w);
-
-    // Normalized screenspace UVs
-    float2 uv = float2((ndc.x + 1.0) * 0.5, (ndc.y + 1.0) * 0.5);
+    const float3 worldPosition = mul(ObjectSrg::GetWorldMatrix(), float4(IN.m_position, 1.0)).xyz;
+    OUT.m_position = mul(ViewSrg::m_viewProjectionMatrix, float4(worldPosition, 1.0));
```

Comment:
suggestion: Is it worth using `.xyz` and then concerting back to a `float4` with `1.0` in the `w` component on the next line? Could it just be?

```
const float3 worldPosition = mul(ObjectSrg::GetWorldMatrix(), float4(IN.m_position, 1.0));
OUT.m_position = mul(ViewSrg::m_viewProjectionMatrix, worldPosition);
```

Sorry if I missed something obvious 😝

Reply:
Your example here
`const float3 worldPosition = mul(ObjectSrg::GetWorldMatrix(), float4(IN.m_position, 1.0));`
wouldn't work as you're assigning the float4 resulting from the mul to a float3.
Did you mean to also change worldPosition to a float4?
Either way the compiler will optimize this so it's not a big deal one way or the other.

Reply:
Ah sorry I meant to write `float4` you're right, so the first line would be...

```
const float4 worldPosition = mul(ObjectSrg::GetWorldMatrix(), float4(IN.m_position, 1.0));
```

It just seemed redundant to me to go to a `float3` and then back to a `float4` but I realize it's not exactly a performance issue, just additional visual clutter